website/views.py

```diff
@@ -451,71 +544,80 @@ def vote_post(request):
     vote_type = request.POST.get('type')
     vote_action = request.POST.get('action')
-    answer_id =  request.POST.get('id') 
+    answer_id = request.POST.get('id')
 
     answer = Answer.objects.get(pk=answer_id)
     cur_post = get_object_or_404(Answer, id=post_id)
 
     userupvote = cur_post.userUpVotes.filter(id=request.user.id).count()
     userdownvote = cur_post.userDownVotes.filter(id=request.user.id).count()
 
-    initial_votes = cur_post.userUpVotes.count() - cur_post.userDownVotes.count()
-
+    initial_votes = cur_post.userUpVotes.count() - cur_post.userDownVotes\
+                                                           .count()
 
     if request.user.id != answer.uid:
 
-        #This loop is for voting
+        # This loop is for voting
         if vote_action == 'vote':
-            if (thisuserupvote == 0) and (thisuserdownvote == 0):
+            if (userupvote == 0) and (userdownvote == 0):
                 if vote_type == 'up':
                     cur_post.userUpVotes.add(request.user)
                 elif vote_type == 'down':
                     cur_post.userDownVotes.add(request.user)
                 else:
                     return HttpResponse("Error: Unknown vote-type passed.")
+            elif (userupvote == 1) and (userdownvote == 0):
+                    if vote_type == 'down':
+                        cur_post.userUpVotes.remove(request.user)
+                        cur_post.userDownVotes.add(request.user)
+            elif (userupvote == 0) and (userdownvote == 1):
+                    if vote_type == 'up':
+                        cur_post.userDownVotes.remove(request.user)
+                        cur_post.userUpVotes.add(request.user)
             else:
                 return HttpResponse(initial_votes)
-        #This loop is for canceling vote
+        # This loop is for canceling vote
         elif vote_action == 'recall-vote':
-            if (vote_type == 'up') and (thisuserupvote == 1):
+            if (vote_type == 'up') and (userupvote == 1):
                 cur_post.userUpVotes.remove(request.user)
-            elif (vote_type == 'down') and (thisuserdownvote == 1):
+            elif (vote_type == 'down') and (userdownvote == 1):
                 cur_post.userDownVotes.remove(request.user)
             else:
                 # "Error - Unknown vote type or no vote to recall"
                 return HttpResponse(initial_votes)
         else:
             return HttpResponse("Error: Bad Action.")
 
-        num_votes = cur_post.userUpVotes.count() - cur_post.userDownVotes.count()
+        num_votes = cur_post.userUpVotes.count() - cur_post.userDownVotes\
+                                                           .count()
         cur_post.num_votes = num_votes
         cur_post.save()
        
-
         return HttpResponse(num_votes)
-            
     
     else:
-    #else:
+        # else:
         return HttpResponse(initial_votes)
 
+
 # Notification Section
```

Comment:
This should be a docstrings. Please refer to Python docs and add this info as a docstring for this function